packages/perseus-editor/src/hint-editor.tsx

```diff
@@ -45,6 +45,15 @@ type HintEditorProps = {
     __type?: "hint";
 };
 
+type DefaultHintEditorProps = {
+    className: HintEditorProps["className"];
+    content: HintEditorProps["content"];
+    replace: HintEditorProps["replace"];
+    showMoveButtons: HintEditorProps["showMoveButtons"];
+    showRemoveButton: HintEditorProps["showRemoveButton"];
+    showTitle: HintEditorProps["showTitle"];
+};
```

Comment:
Question: is this really the type that we want? These properties on `HintEditorProps` are optional and nullable. The corresponding default props should presumably not be nullable.

I'd prefer to keep the old type definition, with explicit types for the properties. The typechecker ensures that the duplicated types stay consistent.

Reply:
I do like the visual connection to the Prop that is being defaulted. However, if I land this PR, I do need to tweak the main HintEditorProps to make the defaulted props non-undefined. Typescript seems smart enough to make them optional as soon as the same key exists in the `static defaultProps`. 

Further, it seems like we might not even need a type for it at all. Any prop that appears in `defaultProps` automatically makes the same-named prop in the component props optional. So maybe typing `defaultProps` is all just yak-shaving and not improving type safety at all.